packages/clients/generic/src/polar-client.ts

```diff
@@ -0,0 +1,181 @@
+      menus: [
+        enabledPlugins.includes('layer-chooser') && {
+          plugin: LayerChooser({}),
+          icon: 'fa-layer-group',
+          id: 'layerChooser',
+        },
+        enabledPlugins.includes('draw') && {
+          plugin: Draw({}),
+          icon: 'fa-pencil',
+          id: 'draw',
+        },
+        enabledPlugins.includes('filter') && {
+          plugin: Filter({
+            layers: {},
+          }),
+          icon: 'fa-filter',
+          id: 'filter',
+        },
+        enabledPlugins.includes('gfi') && {
+          plugin: Gfi({
+            renderType: 'iconMenu',
+            coordinateSources: [],
+            layers: [],
+          }),
+          icon: 'fa-location-pin',
+          id: 'gfi',
+        },
+        enabledPlugins.includes('zoom') && {
+          plugin: Zoom({ renderType: 'iconMenu' }),
+          id: 'zoom',
+        },
+        enabledPlugins.includes('fullscreen') && {
+          plugin: Fullscreen({ renderType: 'iconMenu' }),
+          id: 'fullscreen',
+        },
+        enabledPlugins.includes('geo-location') && {
+          plugin: GeoLocation({
+            renderType: 'iconMenu',
+          }),
+          id: 'geoLocation',
+        },
+        enabledPlugins.includes('attributions') && {
+          plugin: Attributions({
+            renderType: 'iconMenu',
+          }),
+          icon: 'fa-regular fa-copyright',
+          id: 'attributions',
+        },
+      ].filter((x) => x) as Menu[],
+      layoutTag: NineLayoutTag.TOP_RIGHT,
+    })
+
+  coreInstance.addPlugins(
+    [
+      iconMenu,
+      enabledPlugins.includes('address-search') &&
+        AddressSearch({
+          layoutTag: NineLayoutTag.TOP_LEFT,
+        }),
+      enabledPlugins.includes('pins') && Pins({}),
+      enabledPlugins.includes('legend') &&
+        Legend({
+          layoutTag: NineLayoutTag.BOTTOM_RIGHT,
+          displayComponent: true,
+        }),
+      enabledPlugins.includes('export') &&
+        Export({
+          layoutTag: NineLayoutTag.BOTTOM_LEFT,
+          displayComponent: true,
+        }),
+      enabledPlugins.includes('loading-indicator') &&
+        LoadingIndicator({
+          layoutTag: NineLayoutTag.MIDDLE_MIDDLE,
+        }),
+      enabledPlugins.includes('reverse-geocoder') && ReverseGeocoder({}),
+      enabledPlugins.includes('scale') &&
+        Scale({
+          layoutTag: NineLayoutTag.BOTTOM_RIGHT,
+          displayComponent: true,
+        }),
+      enabledPlugins.includes('toast') &&
+        Toast({
+          layoutTag: NineLayoutTag.BOTTOM_MIDDLE,
+          displayComponent: true,
+        }),
+    ].filter((x) => x)
+  )
+}
+
+export default {
+  createMap: ({
+    containerId,
+    services,
+    mapConfiguration,
+    enabledPlugins = [],
+    modifyLayerConfiguration = (x) => x,
+  }: {
+    containerId: string
+    services: object[]
+    mapConfiguration: MapConfig
+    enabledPlugins: Array<PluginName>
+    modifyLayerConfiguration
```

Comment:
This is being introduced by this client and subsequently used in the pages. Do you mind adding some documentation and possibly a type?

Reply:
Whoops. https://github.com/Dataport/polar/pull/91/commits/20fb412fa3267c7d1b0f3f823c50313eca4b5ebc